src/repositories/indexable-repository.php

```diff
@@ -321,13 +332,23 @@ public function find_by_multiple_ids_and_type( $object_ids, $object_type, $auto_
 	 * @return Indexable[] All ancestors of the given indexable.
 	 */
 	public function get_ancestors( Indexable $indexable ) {
-		$hierarchy_table = Yoast_Model::get_table_name( 'Indexable_Hierarchy' );
+		// In the get_ancestors function first do a query purely on the indexable hierarchy table.
+		$ancestors = $this->hierarchy_repository->get_ancestors( $indexable );
+
+		if ( empty( $ancestors ) ) {
+			return [];
+		}
+
+		$indexables = array_column( $ancestors, 'ancestor_id' );
```

Comment:
A reminder of your own comment: `array_column` with an array of objects is only for PHP 7.0+.